File: core/src/main/java/com/ustadmobile/core/contentformats/xapi/Definition.java
package com.ustadmobile.core.contentformats.xapi;

import java.util.List;
import java.util.Map;

public class Definition {

    public Map<String, String> name;

    public Map<String, String> description;

    public String type;

    public Map<String, String> extensions;

    public String moreInfo;

    public String interactionType;

    private List<String> correctResponsePattern;

    public List<Interaction> choices;

    public List<Interaction> scale;

    public List<Interaction> source;

    public List<Interaction> target;

    public List<Interaction> steps;

    private class Interaction {

        private String id;

        private Map<String, String> description;

    }


}

File: core/src/commonMain/kotlin/com/ustadmobile/core/contentformats/xapi/Definition.kt
package com.ustadmobile.core.contentformats.xapi

class Definition {

    var name: Map<String, String>? = null

    var description: Map<String, String>? = null

    var type: String? = null

    var extensions: Map<String, Any>? = null

    var moreInfo: String? = null

    var interactionType: String? = null

    var correctResponsePattern: List<String>? = null

    var choices: List<Interaction>? = null

    var scale: List<Interaction>? = null

    var source: List<Interaction>? = null

    var target: List<Interaction>? = null

    var steps: List<Interaction>? = null

    inner class Interaction {

        var id: String? = null

        var description: Map<String, String>? = null
    }
}
